CloudSQLReplicator/google/cloud/pontem/sql/replicator/util/gcp_api_util.py

```diff
@@ -27,6 +31,112 @@ def _get_user_agent():
   user_agent = 'pontem,{}/{} (gzip)'.format(replicator.__package_name__, replicator.__version__)
   return user_agent
 
+def _get_user_agent_header():
+  """Returns custom User-Agent header."""
+
+  user_agent = _get_user_agent()
+  headers = {'User-Agent': user_agent}
+  return headers
+
+# SQL Admin
+def build_sql_admin_service(credentials=None):
+  """Authorizes and sets custom CloudSQL Replicator user agent.
+
+    Args:
+      project (str): Project ID
+      credentials (google.auth.Credentials): Credentials to authorize client
+
+    Returns:
+      Authorized sqladmin service with CloudSQL Replicator user agent.
+  """
+  headers = _get_user_agent_header()
+  httplib2.Http.request.__func__.func_defaults = ('GET', None, headers, 5, None)
+  default_credentials, _ = google.auth.default()
+  authorized_http = google_auth_httplib2.AuthorizedHttp(
+      credentials or default_credentials
+  )
+
+  service = discovery.build(
+      'sqladmin',
+      'v1beta4',
+      http=authorized_http
+  )
+
+  return service
+
+def create_cloudsql_instance(database_instance_body=None,
+                             project=None,
+                             credentials=None):
+  """Provisions a Cloud SQL instance.
+
+    Args:
+      database_instance_body(JSON): Creation options
+      project(str): Project ID
+      credentials (google.auth.Credentials): Credentials to authorize client
+
+    Returns:
+      JSON: response from sqladmin.instances().insert() call
+  """
+
+  default_credentials, default_project = google.auth.default()
+  default_database_intance_body = {
+    'project': default_project,
+    'name': 'cloudsql-db-{}'.format(uuid.uuid4()),
+    'settings': {
+      'tier': 'db-n1-standard-2'
```

Comment:
not super familiar, but should the tier be a variable?

Reply:
I just set this as the default.  The database_instance_body is a parameter where tier can be passed.

Reply:
Ok, I understand what you are saying now.  I added a constant for default tier.